lib/nats.js

```diff
@@ -627,74 +625,73 @@ Client.prototype.scheduleHeartbeat = function() {
  * @api private
  */
 Client.prototype.setupHandlers = function() {
-    var client = this;
-    var stream = client.stream;
+    const stream = this.stream;
 
     if (undefined === stream) {
         return;
     }
 
-    stream.on('connect', function() {
-        if (client.pingTimer) {
-            clearTimeout(client.pingTimer);
-            delete client.pingTimer;
+    stream.on('connect', () => {
+        if (this.pingTimer) {
```

Comment:
We sure this is safe? Maybe the new syntax makes this ok but I remember that `client=this` was a big deal back in the day.

Reply:
Yes that is the whole thing with the arrow functions - the this there is the one in the outer block, which in this case is the client. It removes all sorts of ambiguities.

Reply:
ok glad they fixed that about the language - lol. I am sure they did a long time ago.. I am slow.